impl/src/metadata.rs

```diff
@@ -50,25 +97,132 @@ pub struct CargoWorkspaceFiles {
+    assert!(files.lock_path_opt.as_ref().map_or(true, |p| p.is_file()));
+
+    // First gather metadata without downloading any dependencies so we can identify any path dependencies.
+    let no_deps_metadata = MetadataCommand::new()
+      .cargo_path(&self.cargo_bin_path)
+      .current_dir(files.toml_path.parent().unwrap()) // UNWRAP: Guarded by function assertion
+      .no_deps()
+      .exec()?;
+
+    // There should be a `Cargo.toml` file in the workspace root
+    fs::copy(
+      no_deps_metadata.workspace_root.join("Cargo.toml"),
+      temp_dir.as_ref().join("Cargo.toml"),
+    )?;
+
+    // Optionally copy over the lock file
+    if no_deps_metadata.workspace_root.join("Cargo.lock").exists() {
+      fs::copy(
+        no_deps_metadata.workspace_root.join("Cargo.lock"),
+        temp_dir.as_ref().join("Cargo.lock"),
+      )?;
+    }
+
+    // Copy over the Cargo.toml files of each workspace member
+    let re = Regex::new(r".+\(path\+file://(.+)\)")?;
+    for member in no_deps_metadata.workspace_members.iter() {
+      // Get a path to the workspace member directory
+      let path = {
+        let capture = match re.captures(&member.repr) {
+          Some(capture) => capture,
+          None => continue,
+        };
+
+        let re_match = match capture.get(1) {
+          Some(re_match) => re_match,
+          None => continue,
+        };
+
+        PathBuf::from(re_match.as_str())
+      };
+
+      let toml_path = path.join("Cargo.toml");
+
+      // If no Cargo.toml file is found, skip this
+      if !toml_path.exists() {
+        continue;
+      }
+
+      // Copy the Cargo.toml files into the temp directory to match the directory structure on disk
+      let diff = diff_paths(&path, &no_deps_metadata.workspace_root).ok_or(anyhow!(
+        "All workspace memebers are expected to be under the workspace root"
+      ))?;
+      let new_path = temp_dir.as_ref().join(diff);
+      fs::create_dir_all(&new_path)?;
+      fs::copy(path.join("Cargo.toml"), new_path.join("Cargo.toml"))?;
+
+      // Additionally, symlink everything in some common source directories to ensure specified
+      // library targets can be relied on and won't prevent fetching metadata
+      for dir in vec!["bin", "src"].iter() {
+        let glob_pattern = format!("{}/**/*.rs", path.join(dir).display());
+        for entry in glob(glob_pattern.as_str()).expect("Failed to read glob pattern") {
+          match entry {
+            Ok(path) => {
+              let diff = diff_paths(&path, &no_deps_metadata.workspace_root).ok_or(anyhow!(
+                "All workspace memebers are expected to be under the workspace root"
+              ))?;
+
+              let new_path = temp_dir.as_ref().join(diff);
+
+              if let Some(parent) = new_path.parent() {
+                fs::create_dir_all(parent)?;
+              }
+
+              self.make_symlink(&path, &new_path)?;
+            },
+            _ => {},
```

Comment:
What situations are caught here? when does this entry not contain Ok?

Reply:
https://docs.rs/glob/0.3.0/glob/ suggests there would be errors in some cases. I opted to ignore them but could escalate that error. Your thoughts?

Reply:
Updated implementation